interfaceUtils.py

```diff
@@ -40,127 +46,139 @@ def requestBuildArea(self):
+        """**Place a block in the world depending on buffer activation**."""
+        if self.__buffering:
+            self.placeBlockBatched(x, y, z, str, self.bufferlimit)
         else:
-            self._setBlockDirect(x, y, z, blockStr)
-
-    def sendOutstandingBlocks(self):
-        self._sendBlocks()
+            self.placeBlock(x, y, z, str)
 
-    def _setBlockDirect(self, x, y, z, blockStr):
-        """**Places a block in the world.**"""
+    def placeBlock(self, x, y, z, str):
+        """**Place a single block in the world**."""
         x, y, z = self.local2global(x, y, z)
 
         url = 'http://localhost:9000/blocks?x={}&y={}&z={}'.format(x, y, z)
-        # DEBUG: print('setting block {} at {} {} {}'.format(blockStr, x, y, z))
         try:
-            response = requests.put(url, blockStr)
+            response = requests.put(url, str)
         except ConnectionError:
             return "0"
         return response.text
-        # DEBUG: print("{}, {}, {}: {} - {}".format(x, y, z, response.status_code, response.text))
 
-    # --------------------------------------------------------- block buffers
+    # ----------------------------------------------------- block buffers
 
-    def _placeBlockBatched(self, x, y, z, blockStr, limit=50):
-        """**Place a block in the buffer and send if the limit is exceeded.**"""
+    def toggleBuffer(self):
+        """**Activates or deactivates the buffer function safely**."""
+        self.Buffering = not self.Buffering
+        return self.Buffering
+
+    @property
+    def Buffering(self):
+        return self.__buffering
+
+    @Buffering.setter
+    def Buffering(self, value):
+        self.__buffering = value
+        if self.__buffering:
+            print("Buffering has been activated.")
+        else:
+            self.sendBlocks()
+            print("Buffering has been deactivated.")
+
+    def placeBlockBatched(self, x, y, z, str, limit=50):
+        """**Place a block in the buffer and send once limit is exceeded**."""
         x, y, z = self.local2global(x, y, z)
 
-        self.buffer.append((x, y, z, blockStr))
+        self.buffer.append((x, y, z, str))
         if len(self.buffer) >= limit:
-            return self._sendBlocks()
+            return self.sendBlocks()
         else:
             return None
 
-    def _sendBlocks(self, x=0, y=0, z=0, retries=5):
-        """**Sends the buffer to the server and clears it.**
+    def sendBlocks(self, x=0, y=0, z=0, retries=5):
+        """**Send the buffer to the server and clear it**.
 
-        Since the buffer contains global coordinates no conversion takes place in this function
+        Since the buffer contains global coordinates
+            no conversion takes place in this function
         """
-
         url = 'http://localhost:9000/blocks?x={}&y={}&z={}'.format(x, y, z)
-        body = str.join("\n", ['~{} ~{} ~{} {}'.format(*bp)
+        body = str.join("\n", ['{} {} {} {}'.format(*bp)
```

Comment:
The relative coordinates were intentional. Leaving the `~` out breaks the relative positioning functionality of sendBlocks, and makes the optional x,y,z parameters useless.